subsys/net/ip/icmpv6.c

```diff
@@ -362,8 +363,12 @@ int net_icmpv6_send_echo_request(struct net_if *iface,
 		return -ENOMEM;
 	}
 
-	net_pkt_set_ip_dscp(pkt, net_ipv6_get_dscp(tc));
-	net_pkt_set_ip_ecn(pkt, net_ipv6_get_ecn(tc));
+	if (priority == -1) {
```

Comment:
```suggestion
	if (priority >= NET_MAX_PRIORITIES) {
		return -EINVAL;
	}
	if (priority < 0) {
```

Reply:
Well, the use case that I am using needs any arbitrary priority to be assigned to the packet, and in all places the if the priority is greater than suported, it falls back to minimal one. e.g., `net_tx_priority2tc`, so, should we allow any priority from the shell?

Reply:
> should we allow any priority from the shell

IMHO we should not allow any arbitrary values to be used here by default. The shell can be used for development purposes but it is mainly meant for end users. If we really want some way to set special values, then I suggest a Kconfig option that is by default 'n' and which enables such behavior.

Reply:
I am fine with adding a Kconfig option `NET_ALLOW_ANY_PRIORITY` that is disabled by default.

Reply:
Sounds good to me.